torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1979,7 +1956,29 @@ void ONNXShapeTypeInference(
     for (auto output : clone_node->outputs()) {
       n_graph->registerOutput(output);
     }
+    // Map original original PyTorch graph's i/o name to temporal ONNX graph's i/o name for shape inference
+    for (size_t i = 0; i < clone_node->inputs().size(); ++i) {
+      torch_to_onnx_input_name[n->input(i)->debugName()] = clone_node->input(i)->debugName();
+    }
 
+    for (size_t i = 0; i < clone_node->outputs().size(); ++i) {
+      torch_to_onnx_output_name[n->output(i)->debugName()] = clone_node->output(i)->debugName();
+    }
+    // Make inferred_shape_data use name from temporal ONNX graph instead of original PyTorch graph
+    // Use original_keys for removing original data which is duplicate
+    std::vector<string> original_keys;
+    for (const auto& gs_data: inferred_shape_data) {
```

Comment:
I think you can create `inferred_shape_data` as empty map. Iterate over `original_shape_data` here, and populate `inferred_shape_data` with temporal_name as key. The benefits are:
* Avoid potential name collision.
* Avoid the need for erase key after this loop.
* Avoid upfront copy cost.

Reply:
Brilliant idea! I have updated it according to your recent comments. Thank you for the review. I will try to resolve lintrunner issue, although it seems unrelated to my PR.